includes/Maps_DisplayMapRenderer.php

```diff
@@ -225,28 +229,35 @@ public static function evilOpenLayersHack( $layers ) {
 				}
 			}
 		}
-
-		MapsMappingServices::getServiceInstance( 'openlayers' )->addLayerDependencies( self::getLayerDependencies( $layerNames ) );
-
 		return $layerDefs;
 	}
 
-	/**
-	 * FIXME
-	 * @see evilOpenLayersHack
-	 */
-	private static function getLayerDependencies( array $layerNames ) {
-		global $egMapsOLLayerDependencies, $egMapsOLAvailableLayers;
+	public static function getLayerDependencies( $service, $params ) {
```

Comment:
I don't see any issues with SM requiring the latest Maps. Why would one want to run an older version anyway? For feature releases at least this makes little sense.

It's really weird that SM uses code from the class responsible for rendering of the displaymap parser function. Presumably something should be factored out.

If we merge the thing in, all of this gets easier to deal with.